tests/utils/test_credentials.py

```diff
@@ -99,3 +102,29 @@ def test_read_mlflow_creds_env_takes_precedence_over_file(tmp_path, monkeypatch)
         creds = read_mlflow_creds()
         assert creds.username == "username_env"
         assert creds.password == "password_env"
+
+
+def test_mlflow_login(tmp_path):
+    with patch("builtins.input") as mock_input, patch("getpass.getpass") as mock_getpass:
+        mock_input.side_effect = ["https://community.cloud.databricks.com/", "weirdmouse"]
+        mock_getpass.side_effect = ["dummypassword"]
+
+        os.environ["DATABRICKS_CONFIG_FILE"] = f"{tmp_path}/.databrickscfg"
```

Comment:
we can use `monkeypatch.setenv` when mocking enviroment variables.

Reply:
i like it, done